script.js

```diff
@@ -0,0 +1,841 @@
+const FUNCTIONAL_KEYS = {
+  shift: false,
+  shiftRight: false,
+  ctrl: false,
+  ctrlRight: false,
+  alt: false,
+  altRight: false,
+  windows: false,
+  caps: false,
+};
+
+if (!window.localStorage.getItem("virtualKeyBoardLang")) window.localStorage.setItem("virtualKeyBoardLang", "en");
+
+const vKeyBoard = {
+  textArea: undefined,
+  focusTextArea: (cursorPosition = vKeyBoard.textArea.selectionStart) => {
+    setTimeout(() => {
+      vKeyBoard.textArea.focus();
+      vKeyBoard.textArea.selectionStart = cursorPosition;
+      vKeyBoard.textArea.selectionEnd = cursorPosition;
+    }, 0);
+  },
+  insertSymbol: (value) => {
+    const cursorPositionStart = vKeyBoard.textArea.selectionStart;
+    const cursorPositionEnd = vKeyBoard.textArea.selectionEnd;
+    vKeyBoard.textArea.value = vKeyBoard.textArea.value.substring(0, cursorPositionStart)
+      + value
+      + vKeyBoard.textArea.value.substring(cursorPositionEnd);
+    vKeyBoard.focusTextArea(cursorPositionStart + 1);
+  },
+};
+
+const body = document.querySelector("body");
+
+const keyboard = document.createElement("div");
+keyboard.className = "keyboard";
+body.appendChild(keyboard);
+
+const textArea = document.createElement("textarea");
+textArea.className = "keyboard__text-area";
+textArea.id = "kbText";
+textArea.autofocus = true;
+textArea.rows = 10;
+vKeyBoard.textArea = textArea;
+keyboard.appendChild(textArea);
+
+const keyboardPanel = document.createElement("div");
+keyboardPanel.className = "keyboard-panel";
+keyboard.appendChild(keyboardPanel);
+
+// structure of key object:
+// {
+//   type: "letter | number | functional",
+//   initial: "value",
+//   shifted: "value when shift is pressed (optional)",
+//   func: "functional keys function",
+//   isClick: "event only on click",
+// };
+
+const arrayOfButtonKeys = [];
+arrayOfButtonKeys.push([

```

Comment:
All this configurations is better to move to another files.

Reply:
Configuration was moved to another file.

https://github.com/BlackTruth/virtual-keyboard/blob/6297a0817c03affb1948c9e73a6352687e6ff449/src/keyboardConstants.js#L1-L8